### PR TITLE
chore: Update kokoro build script to push to Maven

### PR DESCRIPTION
**Pull request by jihuin**

### CODE REVIEW

File: .kokoro/release.sh [hunk 1]
```diff
@@ -1,6 +1,59 @@
...
+  cd function-maven-plugin
+elif [[ $KOKORO_JOB_NAME == *"functions-framework-api"* ]]; then
+  cd functions-framework-api
+else
+  cd invoker
+fi
+echo "pwd=$(pwd)"
+
+# Make sure `JAVA_HOME` is set and using jdk11.
+sudo update-java-alternatives --set java-1.11.0-openjdk-amd64
```
Comment (anniefu) [inline .kokoro/release.sh:57]: Does sudo work within kokoro jobs?

File: .kokoro/release.sh [hunk 2]
```diff
@@ -1,6 +1,59 @@
...
+if [[ $KOKORO_JOB_NAME == *"function-maven-plugin"* ]]; then
+  cd function-maven-plugin
+elif [[ $KOKORO_JOB_NAME == *"functions-framework-api"* ]]; then
+  cd functions-framework-api
+else
+  cd invoker
+fi
+echo "pwd=$(pwd)"
+
+# Make sure `JAVA_HOME` is set and using jdk11.
```
Comment (anniefu) [inline .kokoro/release.sh:56]: [nit] why not java 17?

File: .kokoro/release.sh [hunk 3]
```diff
@@ -1,6 +1,59 @@
...
-echo "JAVA_HOME=$JAVA_HOME"
+# Get secrets from keystore and set and environment variables.
+setup_environment_secrets() {
+  export SONATYPE_USERNAME=functions-framework-release-bot
+  export SONATYPE_PASSWORD=$(cat ${KOKORO_KEYSTORE_DIR}/75669_functions-framework-java-release-bot-sonatype-password)
+  export GPG_PASSPHRASE=$(cat ${KOKORO_KEYSTORE_DIR}/75669_functions-framework-java-release-bot-gpg-passphrase)
+
+  export GNUPGHOME=/tmp/gpg
+  mkdir $GNUPGHOME
+  mv ${KOKORO_KEYSTORE_DIR}/75669_functions-framework-java-release-bot-gpg-pubring $GNUPGHOME/pubring.kbx
```
Comment (anniefu) [inline .kokoro/release.sh:11]: where does this get used? Could you add a comment if this is some magic-path that needs to be correct for something to work.

File: .kokoro/release.sh (hunk 1 above)
Comment (jihuin) [reply .kokoro/release.sh:57]: Yes Kokoro user is added to sudoers file to don't ask password

File: .kokoro/release.sh (hunk 2 above)
Comment (jihuin) [reply .kokoro/release.sh:56]: These artifacts are using jdk11 in the pom file. We can update them and this script later if needed.

File: .kokoro/release.sh (hunk 3 above)
Comment (jihuin) [reply .kokoro/release.sh:12]: Done.

File: .kokoro/release.sh [hunk 4]
```diff
@@ -1,6 +1,60 @@
...
-echo "JAVA_HOME=$JAVA_HOME"
+# Get secrets from keystore and set and environment variables.
+setup_environment_secrets() {
+  export SONATYPE_USERNAME=functions-framework-release-bot
+  export SONATYPE_PASSWORD=$(cat ${KOKORO_KEYSTORE_DIR}/75669_functions-framework-java-release-bot-sonatype-password)
+  export GPG_PASSPHRASE=$(cat ${KOKORO_KEYSTORE_DIR}/75669_functions-framework-java-release-bot-gpg-passphrase)
+
+  # Add the keybox file to $GNUPGHOME to verify the GPG credentials.
+  export GNUPGHOME=/tmp/gpg
+  mkdir $GNUPGHOME
```
Comment (kappratiksha) [inline .kokoro/release.sh:11]: Should we use this https://github.com/GoogleCloudPlatform/appengine-java-standard/blob/main/kokoro/gcp_ubuntu/release.sh#:~:text=%2DDgpg.homedir%3D%24%7BGNUPGHOME%7D

Comment (jihuin) [reply .kokoro/release.sh:11]: Updated. My local test worked without the flags, but it might be needed to twist Kokoro mvn configs

**Review by anniefu — APPROVED** [review]

**Pull request merged**